File: Python/백준/Bronze/19774. ABCD－код/ABCD－код.py
t = int(input())
for i in range(t):
    n = int(input())
    a = n//1000
    b = (n%1000)//100
    c = (n%100)//10
    d = n%10
    p1 = (a*10+b)**2
    p2 = (c*10+d)**2
    if ((p1+p2)%7==1):
        print('YES')
    else:
        print('NO')

File: C#/백준/Bronze/19774. ABCD－код/ABCD－код.cs
using System;

namespace q{
    class w{
        static void Main(){
            int t = int.Parse(Console.ReadLine());
            for (int i=0;i<t;i++){
                int n = int.Parse(Console.ReadLine());
                int a = n/1000;
                int b = n%1000/100;
                int c = n%100/10;
                int d = n%10;
                int p1 = (a*10+b)*(a*10+b);
                int p2 = (c*10+d)*(c*10+d);
                if ((p1+p2)%7==1){
                    Console.WriteLine("YES");
                }
                else{
                    Console.WriteLine("NO");
                }
            }
        }
    }
}